xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1164,6 +1182,53 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     return true;
   }
 
+  absl::Status F8ScaleD(HloInstruction* instr, HloInstruction* existing_gemm,
+                        HloInstruction* d_scale) {
+    if (!ShapeUtil::IsScalar(d_scale->shape())) {
+      return OkStatus();
+    }
+
+    TF_ASSIGN_OR_RETURN(auto gpu_backend_config,
+                        existing_gemm->backend_config<GpuBackendConfig>());
+    const GemmBackendConfig& gemm_backend_config =
+        gpu_backend_config.gemm_backend_config();
+
+    // When the result of an FP8 GEMM is scaled but not type converted to FP8,
+    // the scaling factor must be forwarded to the Custom Call as a_scale
+    // (chosen here) or b_scale. The scaling factor is fused here when no input
```

Comment:
Mention the motivation for using a_sacle is that cublas LT doesn't supoprt having a d_scale when the output is not FP8.